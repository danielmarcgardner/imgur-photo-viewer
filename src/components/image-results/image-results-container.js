import { connect } from 'react-redux';
import ImageResults from './image-results';

export const mapStateToProps = state => ({
  images: state.searchResults.images,
  searchTerm: state.searchResults.searchTerm,
});

export default connect(mapStateToProps)(ImageResults);
