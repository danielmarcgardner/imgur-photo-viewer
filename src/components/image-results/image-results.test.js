import React from 'react';
import { shallow, configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ImageResults from './image-results';
import { mapStateToProps } from './image-results-container';

configure({ adapter: new Adapter() });

const images = [
  {
    image: 'https://i.imgur.com/fZ8JNw4.jpg',
    title: 'Skateboarding Bulldog',
    tags: ['Funny', 'Dog', 'Action'],
    score: 100,
    datetime: 'Mon Oct 02 2017',
    id: 'fZ8JNw4',
  },
  {
    id: 'gzC6hl4',
    image: 'https://i.imgur.com/gzC6hl4.gif',
    title: 'Tired Bulldog Dad Needs Break From His Playful Puppies.',
    score: 39,
    tags: [],
    datetime: 'Mon Oct 02 2017',
  }, {
    id: 'xE7Yu2v',
    image: 'https://i.imgur.com/xE7Yu2v.jpg',
    title: 'Trevor Donovan with his bulldog Tito and German Shepherd Dogbert.',
    score: 26,
    tags: ['dogs', 'dog', 'german she…', 'pets'],
    datetime: 'Mon Oct 02 2017' },
];

describe('ImageResults', () => {
  it('Renders to the page with no images or searchTerm', () => {
    const component = shallow(
      <ImageResults
        searchTerm=""
        images={[]}
      />,
    );
    expect(component).toMatchSnapshot();
  });
  it('Renders to the page with images and searchTerm', () => {
    const component = shallow(
      <ImageResults
        searchTerm="bulldog"
        images={images}
      />,
    );
    expect(component).toMatchSnapshot();
  });
  it('Renders with all the correct properties', () => {
    const component = shallow(
      <ImageResults
        searchTerm="bulldog"
        images={images}
      />,
    );
    expect(component.find('#ImageResults').length).toBe(1);
    expect(component.find('#LoadMoreDiv').length).toBe(1);
  });
  it('mapStateToProps returns the correct state', () => {
    const state = {
      searchResults: {
        images: [
          {
            image: 'https://i.imgur.com/fZ8JNw4.jpg',
            title: 'Skateboarding Bulldog',
            tags: ['Funny', 'Dog', 'Action'],
            score: 100,
            datetime: 'Mon Oct 02 2017',
            id: 'fZ8JNw4',
          },
          {
            id: 'gzC6hl4',
            image: 'https://i.imgur.com/gzC6hl4.gif',
            title: 'Tired Bulldog Dad Needs Break From His Playful Puppies.',
            score: 39,
            tags: [],
            datetime: 'Mon Oct 02 2017',
          }, {
            id: 'xE7Yu2v',
            image: 'https://i.imgur.com/xE7Yu2v.jpg',
            title: 'Trevor Donovan with his bulldog Tito and German Shepherd Dogbert.',
            score: 26,
            tags: ['dogs', 'dog', 'german she…', 'pets'],
            datetime: 'Mon Oct 02 2017' },
        ],
        searchTerm: 'bulldog',
        fetching: false,
        noImages: false,
      },
    };

    const actual = mapStateToProps(state);
    const expected = {
      images: [
        {
          image: 'https://i.imgur.com/fZ8JNw4.jpg',
          title: 'Skateboarding Bulldog',
          tags: ['Funny', 'Dog', 'Action'],
          score: 100,
          datetime: 'Mon Oct 02 2017',
          id: 'fZ8JNw4',
        },
        {
          id: 'gzC6hl4',
          image: 'https://i.imgur.com/gzC6hl4.gif',
          title: 'Tired Bulldog Dad Needs Break From His Playful Puppies.',
          score: 39,
          tags: [],
          datetime: 'Mon Oct 02 2017',
        }, {
          id: 'xE7Yu2v',
          image: 'https://i.imgur.com/xE7Yu2v.jpg',
          title: 'Trevor Donovan with his bulldog Tito and German Shepherd Dogbert.',
          score: 26,
          tags: ['dogs', 'dog', 'german she…', 'pets'],
          datetime: 'Mon Oct 02 2017' },
      ],
      searchTerm: 'bulldog',
    };
    expect(actual).toEqual(expected);
  });
});
