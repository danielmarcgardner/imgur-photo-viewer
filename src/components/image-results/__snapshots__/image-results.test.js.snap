// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageResults Renders to the page with images and searchTerm 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ImageResults
    images={
        Array [
            Object {
              "datetime": "Mon Oct 02 2017",
              "id": "fZ8JNw4",
              "image": "https://i.imgur.com/fZ8JNw4.jpg",
              "score": 100,
              "tags": Array [
                "Funny",
                "Dog",
                "Action",
              ],
              "title": "Skateboarding Bulldog",
            },
            Object {
              "datetime": "Mon Oct 02 2017",
              "id": "gzC6hl4",
              "image": "https://i.imgur.com/gzC6hl4.gif",
              "score": 39,
              "tags": Array [],
              "title": "Tired Bulldog Dad Needs Break From His Playful Puppies.",
            },
            Object {
              "datetime": "Mon Oct 02 2017",
              "id": "xE7Yu2v",
              "image": "https://i.imgur.com/xE7Yu2v.jpg",
              "score": 26,
              "tags": Array [
                "dogs",
                "dog",
                "german sheâ€¦",
                "pets",
              ],
              "title": "Trevor Donovan with his bulldog Tito and German Shepherd Dogbert.",
            },
          ]
    }
    searchTerm="bulldog"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [
          <withRouter()
            id="fZ8JNw4"
            image="https://i.imgur.com/fZ8JNw4.jpg"
/>,
          <withRouter()
            id="gzC6hl4"
            image="https://i.imgur.com/gzC6hl4.gif"
/>,
          <withRouter()
            id="xE7Yu2v"
            image="https://i.imgur.com/xE7Yu2v.jpg"
/>,
        ],
        <div
          id="LoadMoreDiv"
>
          <Connect(Loader) />
</div>,
      ],
      "id": "ImageResults",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "fZ8JNw4",
        "nodeType": "function",
        "props": Object {
          "id": "fZ8JNw4",
          "image": "https://i.imgur.com/fZ8JNw4.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "gzC6hl4",
        "nodeType": "function",
        "props": Object {
          "id": "gzC6hl4",
          "image": "https://i.imgur.com/gzC6hl4.gif",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "xE7Yu2v",
        "nodeType": "function",
        "props": Object {
          "id": "xE7Yu2v",
          "image": "https://i.imgur.com/xE7Yu2v.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": <Connect(Loader) />,
          "id": "LoadMoreDiv",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [
            <withRouter()
              id="fZ8JNw4"
              image="https://i.imgur.com/fZ8JNw4.jpg"
/>,
            <withRouter()
              id="gzC6hl4"
              image="https://i.imgur.com/gzC6hl4.gif"
/>,
            <withRouter()
              id="xE7Yu2v"
              image="https://i.imgur.com/xE7Yu2v.jpg"
/>,
          ],
          <div
            id="LoadMoreDiv"
>
            <Connect(Loader) />
</div>,
        ],
        "id": "ImageResults",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "fZ8JNw4",
          "nodeType": "function",
          "props": Object {
            "id": "fZ8JNw4",
            "image": "https://i.imgur.com/fZ8JNw4.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "gzC6hl4",
          "nodeType": "function",
          "props": Object {
            "id": "gzC6hl4",
            "image": "https://i.imgur.com/gzC6hl4.gif",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "xE7Yu2v",
          "nodeType": "function",
          "props": Object {
            "id": "xE7Yu2v",
            "image": "https://i.imgur.com/xE7Yu2v.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(Loader) />,
            "id": "LoadMoreDiv",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`ImageResults Renders to the page with no images or searchTerm 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ImageResults
    images={Array []}
    searchTerm=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [],
        <div
          id="LoadMoreDiv"
/>,
      ],
      "id": "ImageResults",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": null,
          "id": "LoadMoreDiv",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [],
          <div
            id="LoadMoreDiv"
/>,
        ],
        "id": "ImageResults",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": null,
            "id": "LoadMoreDiv",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
